.github/upload-codecov.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+#  Copyright 2021 PolyCrypt GmbH
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+OUT="target/cov/pallet_perun"
+MERGED="$OUT/merged"
+KCOV="./kcov-build/usr/local/bin/kcov"
+
+rm -rf "$OUT" && mkdir -p "$OUT"
+
+for FILE in conclude deposit dispute unit withdraw; do
+	TEST=$(find target/debug/deps/ -type f -name "$FILE*" -not -name '*.d')
+	[ -x "$TEST" ] || continue
+	echo "Analizing $TEST"
+	$KCOV --exclude-pattern=/.cargo,/usr/lib --verify "$OUT/unmerged-$FILE" "$TEST"
+done
+
+echo "Merging results"
+$KCOV --merge "$MERGED" "$OUT"/unmerged-*
+
+curl -Ls https://codecov.io/bash | bash -s -- -s "$MERGED"
```

Comment:
Why didn't you integrate this script into the workflow file as for example [here](https://github.com/codecov/example-rust)?

If you make this part of the workflow file, you could use the [codecov-action](https://github.com/codecov/codecov-action).

Reply:
It is now part of the workflow and uses the GitHub action for it.